### PR TITLE
Create configmap for starting dispatcher pods

### PR DESCRIPTION
**Pull request by pierDipi**
Create contract configmap for each dispatcher pod as soon as a pod is created, so that even when there are no triggers, kafkasources or subscriptions, the statefulset / pod becomes ready.

Fixes https://github.com/knative-extensions/eventing-kafka-broker/issues/4024
Fixes https://github.com/knative-extensions/eventing-kafka-broker/issues/3995

## Proposed Changes

- Create configmap for starting dispatcher pods

**Release Note**

```release-note

```

**Docs**

### CODE REVIEW

File: control-plane/pkg/reconciler/consumergroup/controller.go [hunk 1]
```diff
@@ -194,6 +198,47 @@ func NewController(ctx context.Context, watcher configmap.Watcher) *controller.I
...
+				if !ok {
+					return false
+				}
+
+				uf := cg.GetUserFacingResourceRef()
+				if uf == nil {
+					return false
+				}
+				if strings.EqualFold(kind, uf.Kind) {
+					return true
```
Comment (Cali0707) [inline control-plane/pkg/reconciler/consumergroup/controller.go:234]: Just so I understand, this is making sure that any consumergroups that are of the same kind (e.g. KafkaSource) are all resynced now that the pod updated?

Comment (pierDipi) [reply control-plane/pkg/reconciler/consumergroup/controller.go:234]: Yes

File: control-plane/pkg/apis/internals/kafka/eventing/data_plane.go [hunk 2]
```diff
@@ -27,6 +27,11 @@ const (
 	ConfigMapVolumeName = "kafka-resources"
 
 	DispatcherVolumeName = "contract-resources"
+
+	DataPlanePodKindLabelKey    = "app.kubernetes.io/kind"
+	DispatcherPodKindLabelValue = "kafka-dispatcher"
```
Comment (creydr) [inline control-plane/pkg/apis/internals/kafka/eventing/data_plane.go:32]: I don't see this label on the dispatcher pods. E.g.: https://github.com/knative-extensions/eventing-kafka-broker/blob/c9e23202d9af3748fe650886569e135b13c7c6d4/data-plane/config/broker/500-dispatcher.yaml#L22-L26. What am I missing?

Comment (pierDipi) [reply control-plane/pkg/apis/internals/kafka/eventing/data_plane.go:32]: Pod label, not statefulset label

https://github.com/knative-extensions/eventing-kafka-broker/blob/c9e23202d9af3748fe650886569e135b13c7c6d4/data-plane/config/broker/500-dispatcher.yaml#L41

Comment (creydr) [reply control-plane/pkg/apis/internals/kafka/eventing/data_plane.go:32]: ah. ok :+1:

Comment (creydr) [reply control-plane/pkg/apis/internals/kafka/eventing/data_plane.go:32]: Thanks for clarifying

**Review by creydr — COMMENTED** [review]
/lgtm

**Comment by raswinraaj** [comment]
Is this fix available in any new stable version or tag?  @pierDipi

**Pull request merged**